plone/app/contenttypes/indexers.py

```diff
@@ -153,6 +153,16 @@ def getObjSize_file(obj):
 
 
 @indexer(IDexterityContent)
+def MimeType(obj):
+    if obj.portal_type in ['File', 'Image']:
```

Comment:
I often have derived own image and file types in my projects. I think it is ok to just call ``aq_base(obj).content_type()'' below and let if fail with an attribute error if it does not apply for the current context.